File: schemas/clusters/level.py
from datetime import datetime
from typing import List, Union
from pydantic import BaseModel

from schemas.clusters.topic import ClusterTopicSchemaBase, ClusterTopicSchemaFull

class ClusterLevelSchemaBase(BaseModel):
    id: int
    cluster_group_id: int
    level_number: int
    cluster_topics: List[Union["ClusterTopicSchemaBase", ClusterTopicSchemaFull]]

    model_config = {
        "from_attributes": True
    }

class ClusterLevelSchemaFull(ClusterLevelSchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend2/types/clusters/level.ts
import type { ClusterTopicSchemaBase, ClusterTopicSchemaFull } from "./topic";

export interface ClusterLevelSchemaBase {
  id: number;
  clusterGroupId: number;
  levelNumber: number;
  clusterTopics: (ClusterTopicSchemaBase | ClusterTopicSchemaFull)[];
}

export interface ClusterLevelSchemaFull extends ClusterLevelSchemaBase {
  createdAt: string;
  updatedAt: string;
}
